src/AnalysisEcce.cxx

```diff
@@ -298,6 +334,9 @@ void AnalysisEcce::Execute()
     // Get the weight for this event's Q2
     auto Q2weightFactor = GetEventQ2Weight(kinTrue->Q2, inLookup[chain->GetTreeNumber()]);
 
+    //fill HFS tree here?
+    if( writeHFSTree && kin->nHFS > 0) HFST->FillTree(1.0);
```

Comment:
Do you want it to be filled for every event or for every hadron in the event? Do you want to add a `weight` branch?

Reply:
Once per event, as it contains all hadrons. Added weight.